src/credentials/providers/ec2CredentialsProvider.ts

```diff
@@ -31,8 +31,13 @@ export class Ec2CredentialsProvider implements CredentialsProvider {
 
         const start = Date.now()
         try {
+            const iamInfo = await this.metadata.getIamInfo()
+            if (!iamInfo || iamInfo.Code !== 'Success') {
+                getLogger().warn(`credentials: missing or invalid instance role: ${iamInfo.Code}`)
```

Comment:
maybe we can be a bit more explicit, to help with support tickets:

```suggestion
                getLogger().warn(`credentials: no role (or invalid) attached to EC2 instance. metadata service /iam/info response: ${iamInfo.Code}`)
```